### PR TITLE
feat(Mapping): Support Mapping.Mapster

### PR DESCRIPTION
**Pull request by zhenlei520**

### CODE REVIEW

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs [hunk 1]
```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Contrib.Data.Mapping.Mapster;
+
+public class DefaultMappingConfigProvider : IMappingConfigProvider
+{
+    private readonly ConcurrentDictionary<(Type SourceType, Type DestinationType, MapOptions? mapOptions), TypeAdapterConfig?>
```
Comment (doddgu) [inline src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs:8]: Why is SourceType pascal, and mapOptions camel?

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    private readonly MapOptions _options;
+
+    public DefaultMappingConfigProvider(MapOptions options) => _options = options;
+
+    public TypeAdapterConfig GetConfig(Type sourceType, Type destinationType, MapOptions? options = null)
+        => GetConfigByCache(sourceType, destinationType, options);
+
+    protected virtual TypeAdapterConfig GetConfigByCache(Type sourceType, Type destinationType, MapOptions? options)
+    {
+        TypeAdapterConfig? config = _store.GetOrAdd(
```
Comment (doddgu) [inline src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs:20]: Why not use lazy or ManualMemoryCache?

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        return new();
+    }
+
+    protected virtual bool IsCollection(Type type)
+        => type.IsGenericType && type.GetInterfaces().Any(x => x.GetGenericTypeDefinition() == typeof(IEnumerable<>));
+
+    protected virtual ConstructorInfo GetBestConstructor(List<ConstructorInfo> destinationConstructors, List<PropertyInfo> sourceProperties)
+    {
+        if (destinationConstructors.Count <= 1)
+            return destinationConstructors.FirstOrDefault()!;
```
Comment (doddgu) [inline src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs:140]: FirstOrDefault()! -> First()

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+    /// <summary>
+    /// Get initial configuration
+    /// When currently in shared mode, return the default global settings
+    /// </summary>
+    /// <returns></returns>
+    protected virtual TypeAdapterConfig GetDefaultConfig(MapOptions? options)
+    {
+        //todo: Other modes are currently not supported, and will be added in the future according to the situation
+        return IsShare(options) ?
```
Comment (doddgu) [inline src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs:188]: ?: -> switch pattern

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/Internal/InvokeBuilder.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Contrib.Data.Mapping.Mapster.Internal;
+
+internal class InvokeBuilder
+{
+    private static readonly MethodInfo NewConfigMethodInfo;
```
Comment (doddgu) [inline src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/Internal/InvokeBuilder.cs:8]: Field name use camal and start with _

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+2. Using `Mapping`
+
+    ```` C#
+    builder.Services.AddMapping();
+    ````
+
+3. Mapping objects
+
+    ````
+    IMapping mapper;// Get through DI
```
Comment (doddgu) [inline src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.md:24]: Mapping? Mapper?

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+    var request = new
+    {
+        Name = "Teach you to learn Dapr...",
+        OrderItem = new OrderItem("Teach you to learn Dapr hand by hand", 49.9m)
+    };
+    var order = mapper.Map<Order>(request);// Map the request to a new object
```
Comment (doddgu) [inline src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.md:31]: Missing the mapping rule description.

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs (hunk 1 above)
Comment (zhenlei520) [reply src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs]: processed

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs (hunk 3 above)
Comment (zhenlei520) [reply src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs]: processed

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs (hunk 4 above)
Comment (zhenlei520) [reply src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs]: processed

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/Internal/InvokeBuilder.cs (hunk 5 above)
Comment (zhenlei520) [reply src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/Internal/InvokeBuilder.cs]: processed

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.md (hunk 6 above)
Comment (zhenlei520) [reply src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.md]: processed

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.md (hunk 7 above)
Comment (zhenlei520) [reply src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.md]: processed

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs (hunk 2 above)
Comment (zhenlei520) [reply src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/DefaultMappingConfigProvider.cs:19]: I don't quite understand what you mean, here is the configuration that tries to get the mapping relationship when the user is mapped

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.zh-CN.md [hunk 8]
```diff
@@ -21,14 +21,14 @@ Masa.Contrib.Data.Mapping.Mapster是基于[Mapster](https://github.com/MapsterMa
...
-    IMapping mapper;// 通过DI获取
+    IMapping mapping;// 通过DI获取
 
     var request = new
     {
         Name = "Teach you to learn Dapr ……",
         OrderItem = new OrderItem("Teach you to learn Dapr hand by hand", 49.9m)
     };
-    var order = mapper.Map<Order>(request);// 将request映射到新的对象
+    var order = mapping.Map<Order>(request);// 将request映射到新的对象, 源对象与目标对象属性名称、类型一致的参数会自动映射、或者目标对象的构造函数参数名称（不区分大小写）、类型与源对象参数一致的，会通过构造函数映射
```
Comment (doddgu) [inline src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.zh-CN.md:31]: Mapping rules require a more detailed description, even a new title

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.zh-CN.md (hunk 8 above)
Comment (zhenlei520) [reply src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.zh-CN.md]: processed

File: test/Masa.Contrib.Data.Mapping.Mapster.Tests/MappingFormTest.cs [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+        };
+        _mapper.Map<CreateUserRequest, User>(request);
+        Assert.IsTrue(TypeAdapterConfig.GlobalSettings.RuleMap.Count == 1);
+    }
+
+    [TestMethod]
+    public void TestAddMultiMapping()
+    {
+        _services.AddMapping();
+        var mappings = _services.BuildServiceProvider().GetServices<IMapper>();
```
Comment (doddgu) [inline test/Masa.Contrib.Data.Mapping.Mapster.Tests/MappingFormTest.cs:24]: mappings -> mapper

File: src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.md (hunk 6 above)
Comment (doddgu) [reply src/Data/Mapping/Masa.Contrib.Data.Mapping.Mapster/README.md]: IMapping?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: test/Masa.Contrib.Data.Mapping.Mapster.Tests/MappingFormTest.cs (hunk 9 above)
Comment (zhenlei520) [reply test/Masa.Contrib.Data.Mapping.Mapster.Tests/MappingFormTest.cs]: processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**